02.async-await/AsyncAwait.Task2.CodeReviewChallenge/Models/Support/ManualAssistant.cs

```diff
@@ -0,0 +1,34 @@
+﻿using System;
+using System.Net.Http;
+using System.Threading;
+using System.Threading.Tasks;
+using CloudServices.Interfaces;
+
+namespace AsyncAwait.Task2.CodeReviewChallenge.Models.Support
+{
+    public class ManualAssistant : IAssistant
+    {
+        private readonly ISupportService _supportService;
+
+        public ManualAssistant(ISupportService supportService)
+        {
+            _supportService = supportService ?? throw new ArgumentNullException(nameof(supportService));
+        }
+
+        public async Task<string> RequestAssistanceAsync(string requestInfo)
+        {
+            try
+            {
+                Task t = _supportService.RegisterSupportRequestAsync(requestInfo);
+                Console.WriteLine(t.Status); // this is for debugging purposes
+                Thread.Sleep(5000); // this is just to be sure that the request is registered
+                return await _supportService.GetSupportInfoAsync(requestInfo)
+                    .ConfigureAwait(false);
+            }
+            catch (HttpRequestException ex)
+            {
+                return await Task.Run(async () => await Task.FromResult($"Failed to register assistance request. Please try later. {ex.Message}"));
```

Comment:
No need to run another task. Just return `Task.FromResult`
This is huge overhead on worker threads. No problem to make sync exit from async method